src/store/user/user.effects.ts

```diff
@@ -201,6 +212,42 @@ export class UserEffects {
       ))
     );
 
+  /**
+   * Get a new access token.
+   */
+  @Effect()
+  refreshAccessToken$ = this.actions$
+    .ofType(UserActions.REFRESH_ACCESS_TOKEN)
+    .withLatestFrom(this.store$)
+    .mergeMap(([action, store]) => Observable.fromPromise(this.storage.get('refresh_token'))
+      .concatMap(token => this.userData.refreshAccessToken({
+          userID: store.user.userID,
+          refreshToken: token
+        })
+        .concatMap(res => Observable.fromPromise(this.storage.set('id_token', res.token))
+          .concatMap(() => [
+            createAction(UserActions.REFRESH_ACCESS_TOKEN_SUCCESS),
+            ...action.payload.success,
+          ])
+          .catch(err => Observable.of(
```

Comment:
This pattern seems repetitive. Could we extract the `catch()` function used in these three places into its own function?